flask_mongorest/resources.py

```diff
@@ -344,6 +357,14 @@ def value_for_field(self, obj, field):
         raise UnknownFieldError
 
     def validate_request(self, obj=None):
+        if (request.method == 'PUT' and obj) or request.method == 'POST':
```

Comment:
Well without this line it breaks tests in main repo (I actually ported that condition from the main repo, maybe that's some corner case we haven't accounted for)